src/AppBundle/Security/TaskVoter.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace AppBundle\Security;
+
+
+use AppBundle\Entity\Task;
+use AppBundle\Entity\User;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class TaskVoter extends Voter
+{
+    public const EDIT = "task_edit";
+    public const DELETE = "task_delete";
+
+    /**
+     * Determines if the attribute and subject are supported by this voter.
+     *
+     * @param string $attribute An attribute
+     * @param mixed $subject The subject to secure, e.g. an object the user wants to access or any other PHP type
+     *
+     * @return bool True if the attribute and subject are supported, false otherwise
+     */
+    protected function supports($attribute, $subject)
+    {
+        if (in_array($attribute, [self::EDIT, self::DELETE]) && $subject instanceof Task) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Perform a single access check operation on a given attribute, subject and token.
+     * It is safe to assume that $attribute and $subject already passed the "supports()" method check.
+     *
+     * @param string $attribute
+     * @param mixed $subject
+     *
+     * @return bool
+     */
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+
+        if (!$user instanceof User) {
+            return false;
+        }
+
+        if (in_array($attribute, [self::EDIT, self::DELETE])) {
+            $task = $subject;
+            if ($user === $task->getAuthor() || in_array("ROLE_ADMIN", $user->getRoles())) {
```

Comment:
Don't search a ROLE in the user roles. You're bypassing role inheritance, use the `AuthorizationCheckerInterface::isGranted` method

Reply:
Ok, it's done.